tools/sdk/ld/eagle.app.v6.common.ld

```diff
@@ -157,6 +157,18 @@ SECTIONS
     *.c.o( EXCLUDE_FILE (umm_malloc.c.o) .literal*, \
            EXCLUDE_FILE (umm_malloc.c.o) .text*)
     *.cpp.o(.literal*, .text*)
+    *.pioenvs\\*\\lib*.a:(EXCLUDE_FILE (umm_malloc.o) .literal*, \
```

Comment:
The same story. In PlatformIO (`platformio.ini), you can have multiple build environment per a single project. For example:
```ini
[env:release]
platform = espressif8266
framework = arduino
board = myboard
build_flags = -DRELEASE ... other flags
lib_deps =
    StableLib@~1.2.3
    otherLib@>2.0

[env:debug]
platform = espressif8266
framework = arduino
board = myboard
build_flags = -DDEBUG... other flags
lib_deps =
    https://git.repo/stable.git
    otherLib@~3.0
```

You will have 2 different build environments and the final objects will be located `%project_root%/.pioenvs/%env_name%/...`. All dependent libraries will be built to static libraries  and placed to `.pioenvs` folder per each build environment.